Create Data Transfer (#65)
iota-community/iota-wiki: docs/learn/about-iota/data-transfer.md

If a node decides that a message is valid it will send this message to it's neighbors using the gossip protocol. Every neighbor that receives the message transfers it again to its neighbors and so on. Very quickly every other node in the network sees the message and has the same information and the same knowledge of the "State" of the network at a given time.

A Message consists of basic information that defines the type and structure of the massage and it can also contain different **Payloads**. A Payload is an attachment that can contain an IOTA transaction, but also all other kinds of data.

The IOTA protocol categorizes these information packages into different types and handles certain types differently than others. Therefore every message that is sent to the network must contain a unique label as an identifier that describes exactly what this message is and what should be done with it. Only with this information "printed" on the 'label', a node will accept and process a message.
This can be understood as similar to having to fill out a label before sending a package using your local postal service.

> A more detailed description of this process on the protocol level can be found here in this document:
> [https://github.com/iotaledger/protocol-rfcs/blob/33570042d8f2241ecc9a9104f5eb38fbf4bc3c95/text/0017-message/0017-message.md](https://)

Reviewer: ```suggestion
A more detailed description of this process on the protocol level can be found [here](https://github.com/iotaledger/protocol-rfcs/blob/33570042d8f2241ecc9a9104f5eb38fbf4bc3c95/text/0017-message/0017-message.md).
```

        <td colspan="2">
            <details open="true">
                <summary>Generic Payload</summary>
                <blockquote>
                An outline of a general payload
                </blockquote>
                <table>
                    <tr>
                        <th>Name</th>
                        <th>Type</th>
                        <th>Description</th>
                    </tr>
                    <tr>
                        <td>Payload Type</td>
                        <td>uint32</td>
                        <td>
                            The type of the payload. It will instruct the node on how to parse the fields that follow.
                        </td>
                    </tr>
                    <tr>
                        <td>Data Fields</td>
                        <td>ANY</td>
                        <td>A sequence of fields, where the structure depends on <code>payload type</code>.</td>
                    </tr>
                </table>
            </details>
    <tr>
        <td>Nonce</td>
        <td>uint64</td>
        <td>The nonce which lets this message fulfill the Proof-of-Work requirement.</td>
    </tr>
</table>

> All this information must be created by a client (wallet software or other programs that generate IOTA messages) to be able to issue a message to a node and make sure the node knows what to do with this message.

Reviewer: I think this is where I would link to the technical documentation for details. The part preceding this part already does an amazing job explaining what to expect in a message in a beginner friendly way.

Author: yeah agree - that was actually the plan... if someone want the total details they will be in the docs. i hope they get managed to structurize the RFC process into propper documents...

---
title: Data Transfer
description: A major use-case for the IOTA Technology is Data Transfer. Learn how it works.
---

# Data Transfer

> IOTA is the only Cryptocurrency and DLT Technology that is capable of transferring Data totally free, fast, immutable, unforgeable, and secure. 

Reviewer: I like these bold statements, but they might get outdated pretty quickly as other projects develop. Just leaving this comment here as reminder to discuss on Discord.

> These unique feautures open up a wide array of use cases that normal cryptocurrencies can never support.

Reviewer: ```suggestion
These unique features open up a wide array of use cases that normal cryptocurrencies can never support.
```

Again a bold statement that we might want to avoid. My opinion would be to base our information on long lasting facts, rather than current state that might change quickly.

Author: removed and rewritten

The economies of the future will rely on access to Data and on ways to monetize this Data. IOTA offers the solutions in its core protocol to transform societies worldwide into Data Economies.

> In IOTA, Nodes send and receive messages (data objects). These messages can contain value - in form of an IOTA token transaction - but they don't need to. 

Reviewer: ```suggestion
In IOTA, Nodes send and receive messages (data objects). There are several types of messages, like transaction messages which transfer value and optionally data, but also data only messages which enable free data transfer. 
```
